### PR TITLE
thorvg: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **thorvg/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
@RubenRBS 
Would you please assist me as I don't have many recipes that I can refer to regarding meson?
I seem to get an error with the following code when I specify a BACKEND to MensonToolchain.
````python
meson = Meson(self)
meson.install()
````

The error is as follows.
```
... \src\meson.build:1:0: ERROR: Tried modify read only option 'backend'
```

Is there a problem with the way my conanfile.py is written and is there a workaround?

**Comment by RubenRBS** [comment]
Hi @toge, I've pinged @franramirez688 which is more faimiliar than I am with Meson to take a look into this, thanks for the contribution as always, let's see what they find about your error :)

**Comment by toge** [comment]
@RubenRBS 
Thanks a lot!

**Comment by danimtb** [comment]
@toge FYI, I was having a look at this with @franramirez688 and we found a bug for the `backend='vs'` in Meson during the install step in `package()` method. We will report it upstream to get it fixed in new releases.

Meanwhile, we realized that using `backend='vs2022'` worked without issue, so let's try that in the recipe for now 😄 Thank you!

**Comment by franramirez688** [comment]
> @toge FYI, I was having a look at this with @franramirez688 and we found a bug for the `backend='vs'` in Meson during the install step in `package()` method. We will report it upstream to get it fixed in new releases.
> 
> Meanwhile, we realized that using `backend='vs2022'` worked without issue, so let's try that in the recipe for now 😄 Thank you!

https://github.com/mesonbuild/meson/issues/12868 cc @toge @RubenRBS @danimtb

**Comment by toge** [comment]
@danimtb 
Thanks a lot!
I will try to fix it.

**Comment by franramirez688** [comment]
Hi @toge 

I've been working on this https://github.com/conan-io/conan/pull/15706 
That PR is removing the `reconfigure` process, so we shouldn't have any problem keeping the original `vs` backend. I think we can wait for it and try this PR again using Conan's latest version.

**Comment by toge** [comment]
@franramirez688 
Thank you for information!
I'm waiting for it.

File: recipes/thorvg/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    def generate(self):
+        if is_msvc(self):
+            tc = MesonToolchain(self, backend="vs")
+        else:
+            tc = MesonToolchain(self)
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        venv = VirtualBuildEnv(self)
+        venv.generate(scope="build")
```
Comment (franramirez688) [inline recipes/thorvg/all/conanfile.py:86]:
```suggestion
        venv.generate()
```

File: recipes/thorvg/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.scm import Version
+from conan.tools.microsoft import is_msvc
+from conan.tools.env import VirtualBuildEnv
+import os
+
+
+required_conan_version = ">=1.53.0"
```
Comment (franramirez688) [inline recipes/thorvg/all/conanfile.py:15]:
```suggestion
required_conan_version = ">=1.64.0 <2 || >=2.2.0"
```

**Review by franramirez688 — COMMENTED** [review]
Minor suggestions, otherwise, it looks great.

File: recipes/thorvg/all/conanfile.py (hunk 2 above)
Comment (franramirez688) [reply recipes/thorvg/all/conanfile.py:15]: This PR needs Conan 1.64 version

cc @danimtb

**Review by franramirez688 — DISMISSED** [review]

**Comment by toge** [comment]
@franramirez688 
Thank you for your review and updating this PR!
I am waiting for updating conan 1.64.0 in CCI.

File: recipes/thorvg/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            self.tool_requires("pkgconf/2.1.0")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        if is_msvc(self):
+            tc = MesonToolchain(self, backend="vs")
+        else:
+            tc = MesonToolchain(self)
```
Comment (franramirez688) [inline recipes/thorvg/all/conanfile.py:81]: @toge I just realized about the available options https://github.com/thorvg/thorvg/blob/main/meson_options.txt
We should take care of them.

**Review by franramirez688 — CHANGES_REQUESTED** [review]

**Review by franramirez688 — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Comment by toge** [comment]
@franramirez688 
Thanks a lot for lots of improvements!

**Pull request merged**